tests/CredentialSource/FileSourceTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Tests\CredentialSource;
+
+use Google\Auth\CredentialSource\FileSource;
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use UnexpectedValueException;
+
+/**
+ * @group credentialsource
+ * @group credentialsource-file
+ */
+class FileSourceTest extends TestCase
```

Comment:
Can you add tests for the happy path for both json and text format?

Reply:
Oh whoops, I did understand how these tests were running with the @dataProvider, I see now, my bad!